intel-lustre-clients-vmss-centos/azuredeploy.json

```diff
@@ -0,0 +1,414 @@
+      }
+    },
+    "clientVmSize": {
+      "type": "string",
+      "defaultValue": "Standard_D2",
+      "metadata": {
+        "description": "Size of the Lustre client VM"
+      },
+      "allowedValues": [
+        "Standard_D2",
+        "Standard_D3",
+        "Standard_D4",
+        "Standard_D11",
+        "Standard_D12",
+        "Standard_D13",
+        "Standard_D14",
+        "Standard_D2_V2",
+        "Standard_D3_V2",
+        "Standard_D4_V2",
+        "Standard_D5_V2",
+        "Standard_D11_V2",
+        "Standard_D12_V2",
+        "Standard_D13_V2",
+        "Standard_D14_V2",
+        "Standard_DS2",
+        "Standard_DS3",
+        "Standard_DS4",
+        "Standard_DS11",
+        "Standard_DS12",
+        "Standard_DS13",
+        "Standard_DS14",
+        "Standard_G1",
+        "Standard_G2",
+        "Standard_G3",
+        "Standard_G4",
+        "Standard_G5",
+        "Standard_GS1",
+        "Standard_GS2",
+        "Standard_GS3",
+        "Standard_GS4",
+        "Standard_GS5",
+        "Standard_A2",
+        "Standard_A3",
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_A6",
+        "Standard_A7",
+        "Standard_A8",
+        "Standard_A9",
+        "Standard_A1",
+        "Standard_D1",
+        "Standard_DS1"
+      ]
+    },
+    "clientCount": {
+      "type": "int",
+      "defaultValue": 2,
+      "minValue": 1,
+      "maxValue": 100,
+      "metadata": {
+        "description": "Number of Lustre client instances"
+      }
+    },
+    "filesystemName": {
+      "type": "string",
+      "defaultValue": "scratch",
+      "metadata": {
+        "description": "Name of the Lustre filesystem exposed by the Lustre MGS node"
+      }
+    },
+    "mgsIpAddress": {
+      "type": "string",
+      "defaultValue": "10.1.0.4",
+      "metadata": {
+        "description": "IP address of the Lustre MGS node"
+      }
+    },
+    "existingVnetResourceGroupName": {
+      "type": "string",
+      "metadata": {
+        "description": "Existing Virtual Network Resource Group where Lustre servers are deployed"
+      }
+    },
+    "existingVnetName": {
+      "type": "string",
+      "defaultValue": "vnet-lustre",
+      "metadata": {
+        "description": "Existing Virtual Network name (e.g. vnet-lustre)"
+      }
+    },
+    "existingSubnetClientsName": {
+      "type": "string",
+      "defaultValue": "subnet-lustre-clients",
+      "metadata": {
+        "description": "Lustre clients will be deployed into this subnet within the existing Virtual Network"
+      }
+    }
+  },
+  "variables": {
+    "apiVersion": "2015-06-15",
+    "storageAccountType": "Standard_LRS",
+    "saCount": 5,
+    "newStorageAccountSuffix": "[concat(parameters('vmssName'), 'sa')]",
+    "uniqueStringArray": [
+      "[concat(uniqueString(concat(resourceGroup().id, deployment().name, variables('newStorageAccountSuffix'), '0')))]",
```

Comment:
There was a bug report a few months ago that said have a deployment().name as a seed in uniquestring was a bad idea for testing and deploying incremental resources into the same environment. Each time you specified a different deployment name while adding resources to the same template, it would result in different storage accounts to be spun up.